### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by broomburgo**

### CODE REVIEW

**Comment by broomburgo** [comment]
ops

**Comment by broomburgo** [comment]
@bkase @CodaFi I added a few instructions in the README for using `Abstract` in projects that use Carthage: `Operadics` must be built manually, so I added a bash script to the repository for doing that after the first `carthage update` fails.

**Comment by CodaFi** [comment]
If you `--use-submodules` then I believe Carthage won't thrash like this.

**Comment by broomburgo** [comment]
`--use-submodule` will add the checked out project to git as a submodule, but it won't change the behavior of `update` or `bootstrap`, that'll still include a build phase that's going to fail.

The most common use case for Carthage is to simply call `carthage update` which also builds the dependencies, but if a project depends on `Abstract`, or even worse, if `Abstract` is a second-level dependency via another project (that's probably going to be more frequent, actually), the build part will fail. Adding a script to build `Operadics` and a few lines of README seems like a decent solution if we don't add Carthage compatibility to `Operadics`.

**Comment by CodaFi** [comment]
I don't understand why this is happening given that we use Operadics in the other carthage builds (Swiftz and Swiftx) and don't seem to be subject to this problem.

**Comment by broomburgo** [comment]
I think I cracked it... in Swiftx and Swiftz, the Operadics repository is a submodule, which means that when cloning those repos the `--recursive` flag is needed for the project to be buildable. This is a minor hassle when working within Abstract, but users of the library should have no problem with Carthage because `carthage update` will also recursively clone all the dependencies.  I'm going to do the same for Abstract and update the pr.

**Comment by broomburgo** [comment]
I think it should be fine now.

File: Tests/AbstractTests/Utility/CustomArbitraryTypes.swift [hunk 1]
```diff
@@ -1,8 +1,5 @@
 import XCTest
 @testable import Abstract
-#if !XCODE_BUILD
```
Comment (CodaFi) [inline Tests/AbstractTests/Utility/CustomArbitraryTypes.swift:3]: This is needed for the package build still, no?  You can use the positive condition `#if SWIFT_PACKAGE` and remove `XCODE_BUILD` from the project if you wish.

Comment (broomburgo) [reply Tests/AbstractTests/Utility/CustomArbitraryTypes.swift:3]: You're right, I just changed everything into SWIFT_PACKAGE and removed the XCODE_BUILD flag

**Review by CodaFi — APPROVED** [review]

**Pull request merged**